lua/zipkin.lua

```diff
@@ -65,39 +65,46 @@ function zipkin.inject_zipkin_headers(incoming_zipkin_headers)
     return headers
 end
 
--- If Zipkin headers exist and the request is sampled, then log them to syslog.
+-- If Zipkin headers exist, then log them to syslog.
 -- Start and end times are in epoch seconds, but Zipkin wants them in
 -- microseconds.
 function zipkin.emit_syslog(headers, start_time, end_time)
-    local sampled = headers['X-B3-Sampled']
-    local request_string = string.format('"%s %s %s"',
-        ngx.var.request_method,
-        ngx.var.request_uri,
-        ngx.var.server_protocol
-	)
-    local message = string.format(
-        'spectre/zipkin %s %s %s %s %s %d %d, client: %s, server: , request: %s',
-        headers['X-B3-TraceId'],
-        headers['X-B3-SpanId'],
-        headers['X-B3-ParentSpanId'],
-        headers['X-B3-Flags'],
-        sampled,
-        start_time * 1000000,
-        end_time * 1000000,
-        ngx.var.remote_addr,
-        request_string
-    )
-
-	-- RFC5424 is the syslog format. We encode the message and send it along to syslog2scribe
-    local encoded_message = rfc5424.encode(
-        "LOCAL0",
-        "INFO",
-        ngx.var.hostname,
-        ngx.var.pid,
-        "nginx_spectre",
-        message
-    )
-    sock:send(encoded_message)
+    if headers['X-B3-TraceId'] ~= nil and
+            headers['X-B3-SpanId'] ~= nil and
+            headers['X-B3-ParentSpanId'] ~= nil and
+            headers['X-B3-Flags'] ~= nil and
+            headers['X-B3-Sampled'] ~= nil then
```

Comment:
I think `X-B3-Sampled` is not required. Also, we probably just want to log a placeholder here since some requests (e.g. pagelets) will not have spanids, but it might be still useful to have them in the logs

Reply:
I think that if `X-B3-Sampled` is not there syslog2scribe will ignore the line. Unless we want to default it to 0.

Not sure what you mean about logs, but those logs don't go anywhere except zipkin

Reply:
I meant that whatever is consuming these log lines (e.g. syslog2scribe internally) can still use it.

In fact `X-B3-Sampled` missing should still cause firehose spans. In the B3 specification, this value missing simply means a deferred sampling decision. But it's irrelevant in firehose tracing mode.

Perhaps we need to adjust syslog2scribe to handle this fact.

BTW, this is fact holds for `X-B3-Flags` as well. It is not required. https://github.com/openzipkin/b3-propagation#options